photon/libphoton/photon/photon-tile.hh

```diff
@@ -105,14 +106,30 @@ struct TileEmu {
     UNARY_OP(vsigmoid_u8,  uint8_t, 64, arithmetic::sigmoid_u8)
     UNARY_OP(vsigmoid_fp16, half, 32, arithmetic::sigmoid_fp16)
 
-    inline void vld(uint8_t * addr, VREG_T& dst) {
-        hooks->on_mem_read(tile_id, (uintptr_t)addr);
-        memcpy(dst.bytes, addr, 64);
+    static inline void vld(uint8_t * addr, VREG_T& dst) {
+        assert(64 % BLOCK_SIZE == 0);
+        for (size_t i = 0; i < 64; i += BLOCK_SIZE) {
+            addr += i;
+            uint8_t * cl_addr = Cache::read(addr);
+            memcpy(dst.bytes + i, cl_addr, BLOCK_SIZE);
+        }
     }
 
     static inline void vst(uint8_t * addr, VREG_T& dst) {
         memcpy(addr, dst.bytes, 64);
     }
+
+    static inline void prefetch(uint8_t * addr, size_t len) {
+        size_t i = 0;
+        for (i = len; i > BLOCK_SIZE; i -= BLOCK_SIZE) {
+            Cache::fetch(addr);
+            addr += BLOCK_SIZE;
+        }
+
+        if(i > 0) {
+            Cache::fetch(addr);
+        }
```

Comment:
So since the cache hierarchy is functionally transparent to the user (aside from PREFETCH1 instruction), this function can be a no-op. 

I see what you're trying to do though so I'll suggest the following:
Have a hook here called `on_prefetch` (see https://github.com/VerticalResearchGroup/eagle/pull/5). Some other location should be responsible for adding a listener on this hook to update a tile's private cache state (tags).